databricks/koalas/series.py

```diff
@@ -5590,11 +5593,12 @@ def compare(
         2    c     c
         3    d     b
         4    e     e
+
+        >>> reset_option("compute.ops_on_diff_frames")
         """
-        with option_context("compute.ops_on_diff_frames", True):
-            if not self.index.equals(other.index):
-                raise ValueError("Can only compare identically-labeled Series objects")
-            combined = combine_frames(self.to_frame(), other.to_frame())
+        if not self.index.equals(other.index):
+            raise ValueError("Can only compare identically-labeled Series objects")
+        combined = combine_frames(self.to_frame(), other.to_frame())
```

Comment:
Btw, we should consider other is from the same anchor:

```py
>>> s1 = ks.Series(["a", "b", "c", "d", "e"])
>>> s1.compare(s1)
Traceback (most recent call last):
...
AssertionError: We don't need to combine. `this` and `that` are same.
```

Reply:
Good idea! Let me file a new PR for this.